celery/loaders/base.py

```diff
@@ -268,13 +267,6 @@ def find_related_module(package, related_name):
             raise
 
     try:
-        pkg_path = importlib.import_module(package).__path__
-    except AttributeError:
-        return
-
-    try:
-        _imp.find_module(related_name, pkg_path)
+        return importlib.import_module('{0}.{1}'.format(package, related_name))
     except ImportError:
```

Comment:
I think you do not need to account for `AttributeError` anymore. The failing test case can be re-formatted as follows:

```
def test_find_related_module(self):
        with patch('importlib.import_module') as imp:
            imp.return_value = Mock()
            imp.return_value.__path__ = 'foo'
            base.find_related_module(base, 'tasks')

            imp.side_effect = ImportError()
            base.find_related_module('celery.loaders.base', 'tasks')

```